festflow/views.py

```diff
@@ -109,3 +112,30 @@ def complete_profile(request):
     context['backend'] = backend
     context['profile_form'] = profile_form
     return render(request, 'festflow/complete_profile.html', context)
+
+
+def subscribe(request):
+    context = {}
+
+    if request.method == 'POST':
+        form = SubscriptionForm(request.POST)
+        to_addr = request.POST.get('contact_email', '')
+        if Subscription.objects.filter(contact_email=to_addr).count():
```

Comment:
filtering is not efficient. Instead just use get() [1]. And apply index=True in the email id field in Subscriptions. That will be faster. Do you agree?

[1] http://stackoverflow.com/questions/639836/what-is-the-right-way-to-validate-if-an-object-exists-in-a-django-view-without-r